### PR TITLE
perf: improve application toggle flow

### PR DESCRIPTION
**Pull request by anonrig**
This pull request includes some caching on translation logic. I've also added some TODOs for future reference.

### CODE REVIEW

File: packages/trpc/server/routers/viewer/apps.tsx [hunk 1]
```diff
@@ -142,6 +142,8 @@ export const appsRouter = router({
 
       // If disabling an app then we need to alert users based on the app type
       if (input.enabled) {
```
Comment (roae) [inline packages/trpc/server/routers/viewer/apps.tsx:144]: disabling or enabling?

File: packages/trpc/server/routers/viewer/apps.tsx [hunk 2]
```diff
@@ -142,6 +142,8 @@ export const appsRouter = router({
...
+        const translations = new Map(["en", await getTranslation("en", "common")]);
```
Comment (roae) [inline packages/trpc/server/routers/viewer/apps.tsx:145]: English will always be loaded, even when the user does not use it?

File: packages/trpc/server/routers/viewer/apps.tsx [hunk 3]
```diff
@@ -158,18 +160,28 @@ export const appsRouter = router({
             },
           });
 
+          // TODO: This should be done async probably using a queue.
           Promise.all(
             appCredentials.map(async (credential) => {
```
Comment (roae) [inline packages/trpc/server/routers/viewer/apps.tsx:165]: Here we could use `for ..of`, right?

**Review by roae — CHANGES_REQUESTED** [review]

File: packages/trpc/server/routers/viewer/apps.tsx (hunk 3 above)
Comment (anonrig) [reply packages/trpc/server/routers/viewer/apps.tsx:165]: For of will block the execution. Cal needs something like a queue that handles this operation async and unrelated from the execution of this current function.

File: packages/trpc/server/routers/viewer/apps.tsx (hunk 2 above)
Comment (anonrig) [reply packages/trpc/server/routers/viewer/apps.tsx]: Updated with the changes.

**Review by zomars — APPROVED** [review]
Pushed some fixes, tested and LGTM to ship.

LGTM

File: packages/trpc/server/routers/viewer/apps.tsx (hunk 1 above)
Comment (zomars) [reply packages/trpc/server/routers/viewer/apps.tsx]: pushed a fix for this

**Review by roae — APPROVED** [review]
LGTM!

**Pull request merged**